src/region_growing/label_connected_comp.py

```diff
@@ -103,6 +117,48 @@ def _fill_components(self, threshold=0.1):
 
         return label_mask, points_added
 
+    def _fill_car_like_components(self, road_polygons, ahn_tile):
+        """ Label car like clusters.  """  # TODO text
+        mask_indices = np.where(self.mask)[0]
+        label_mask = np.zeros(len(self.mask), dtype=bool)
+
+        cc_labels, counts = np.unique(self.point_components,
+                                      return_counts=True)
+
+        cc_labels_filtered = cc_labels[counts >= self.min_component_size]
+
+        surface = ahn_tile['ground_surface']
+        fast_z = FastGridInterpolator(ahn_tile['x'], ahn_tile['y'], surface)
+
+        for cc in cc_labels_filtered:
+            # select points that belong to the cluster
+            cc_mask = (self.point_components == cc)
+
+            target_z = fast_z(self.las[mask_indices[cc_mask]])
+            valid_values = target_z[np.isfinite(target_z)]
+
+            if valid_values.size != 0:
+                max_z_thresh = np.mean(valid_values) + self.max_above_ground
+
+                max_z = np.amax(self.las[mask_indices[cc_mask]][:,2])  # TODO miss de cc cloud gebruiken?
+                if max_z < max_z_thresh:
+                    hull_points, mbr_width, mbr_length = minimum_bounding_rectangle(self.las[mask_indices[cc_mask]][:,:2])
+
+                    if self.min_width_thresh < mbr_width < self.max_width_thresh and self.min_length_thresh < mbr_length < self.max_length_thresh: # TODO too long
+                        p1 = Polygon(hull_points)
+                        for road_polygon in road_polygons:
+                            p2 = Polygon(road_polygon)
+
+                            do_overlap = p1.intersects(p2)
+                            if do_overlap:
```

Comment:
Misschien werkt het beter om niet een minimale intersect goed te keuren, maar alleen als de overlap "groot genoeg" is?